ocs_ci/ocs/warp.py

```diff
@@ -0,0 +1,149 @@
+
+    def create_resource_warp(self):
+        """
+        Create resource for Warp S3 benchmark:
+            * Create service account
+            * Create PVC
+            * Create warp pod for running workload
+
+        """
+
+        # Create service account
+        self.sa_name = helpers.create_serviceaccount(self.namespace)
+        self.sa_name = self.sa_name.name
+        helpers.add_scc_policy(sa_name=self.sa_name, namespace=self.namespace)
+
+        # Create test pvc+pod
+        log.info(f"Create Warp pod to generate S3 workload in {self.namespace}")
+        pvc_size = "50Gi"
+        self.pod_name = "warp-pod"
+        self.pvc_obj = helpers.create_pvc(
+            sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+            namespace=self.namespace,
+            size=pvc_size,
+        )
+        self.pod_obj = helpers.create_pod(
+            constants.CEPHBLOCKPOOL,
+            namespace=self.namespace,
+            pod_name=self.pod_name,
+            pvc_name=self.pvc_obj.name,
+            sa_name=self.sa_name,
+            pod_dict_path=self.pod_dic_path,
+            dc_deployment=True,
+            deploy_pod_status=constants.STATUS_COMPLETED,
+        )
+
+    def install_warp(self, timeout=1800):
+        """
+        Install minio warp S3 benchmark:
+        https://github.com/minio/warp
+
+        """
+        # Install warp
+        log.info(f"Installing warp S3 benchmark on testing pod {self.pod_obj.name}")
+        warp_path = "https://github.com/minio/warp/releases/download/v0.5.5/warp_0.5.5_Linux_x86_64.tar.gz"
+        self.pod_obj.exec_cmd_on_pod(f"wget {warp_path}", timeout=timeout)
+        self.pod_obj.exec_cmd_on_pod(
+            "tar -xzvf warp_0.5.5_Linux_x86_64.tar.gz", timeout=timeout
+        )
+        log.info("Successfully installing warp s3 benchmark")
+
+    def run_benchmark(
+        self,
+        bucket_name=None,
+        access_key=None,
+        secret_key=None,
+        duration=None,
+        concurrent=None,
+        objects=None,
+        obj_size=None,
+        timeout=None,
+    ):
+        """
+         Running Warp S3 benchmark
+         Usage detail can be found at: https://github.com/minio/warp
+
+        Args:
+            bucket_name (string): Name of bucket
+            access_key (string): Access Key credential
+            secret_key (string): Secret Key credential
+            duration (string): Duration of the test
+            concurrent (int): number of concurrent
+            objects (int): number of objects
+            obj_size (int): size of object
+            timeout (int): timeout in seconds
+
+        """
+
+        # Running warp S3 benchmark
+        log.info("Running Minio Warp S3 benchmark")
+        timeout = timeout if timeout else 3600
+        self.access_key = access_key if access_key else self.access_key
+        self.secret_key = secret_key if secret_key else self.secret_key
+        self.bucket_name = bucket_name if bucket_name else self.bucket_name
```

Comment:
there is no `self.bucket_name` in the init!!

Reply:
Added self.bucket_name in the init.

Reply:
If you pass None as bucket_name, i guess it will throw.. bucket not found error. It's rather better to have a logic, if `bucket_name==None` then let it create the default bucket (which warp takes care of it)

Reply:
Bucket creation will be happened before the test run and passing bucket name to use in the test execution:
https://github.com/tiffanyn108/ocs-ci/blob/scale_nb_endpoint/tests/e2e/scale/noobaa/test_warp.py#L49

Reply:
Yes it works fine with your test, but what if someone in future uses  your code and doesn't pass bucket_name (as warp can create a default bucket).. I feel its better to address that feature of warp as well..

Reply:
The option to specify the bucket name while running warp will give you more control on which bucket you want to run on instead of warp create default bucket on the fly.